src/bloqade/analysis/address/impls.py

```diff
@@ -190,7 +190,7 @@ def unwrap(
         stmt: squin.wire.Unwrap,
     ):
 
-        origin_qubit = frame.get(stmt.qubit)
+        origin_qubit = frame.get_casted(stmt.qubit, AddressQubit)
 
         return (AddressWire(origin_qubit=origin_qubit),)
```

Comment:
```suggestion
        origin_qubit = frame.get(stmt.qubit)
        
        if isintance(origin_qubit, AddressQubit):

            return (AddressWire(origin_qubit=origin_qubit),)
        else:
            return (Address.top(), )
```

I think the implementation of this function is wrong maybe @johnzl-777 can confirm if this makes sense?

Reply:
Yes, it does! Thank you for catching this (and explaining how to handle these pyright complaints moving forward 👍 )